### PR TITLE
Verifying only one final per championship

### PR DESCRIPTION
**Pull request by cristiadu**

### CODE REVIEW

File: Frontend/src/app/insert-match/insert-match.component.ts [hunk 1]
```diff
@@ -61,6 +61,12 @@ export class InsertMatchComponent implements OnInit, OnChanges {
 			this.error = {description: 'Verifique se há um campeonato criado.'};
 			return false;
 		}
+
+		if (this.matchService.getFinals() && this.match.isFinal) {
```
Comment (aquintiliano) [inline Frontend/src/app/insert-match/insert-match.component.ts]: I think you didn't understand how matchService.getFinals() work

**Review by aquintiliano — DISMISSED** [review]

File: Frontend/src/app/insert-match/insert-match.component.ts (hunk 1 above)
Comment (cristiadu) [reply Frontend/src/app/insert-match/insert-match.component.ts]: I did, I just thought empty arrays would be false if checked as booleans.. Just a small change and it's ready to go.

Comment (cristiadu) [reply Frontend/src/app/insert-match/insert-match.component.ts]: In fact yeah, I didn't haha. But fixed now.

**Comment by cristiadu** [comment]
Addressed comments

File: Frontend/src/app/insert-match/insert-match.component.ts [hunk 2]
```diff
@@ -95,16 +102,31 @@ export class InsertMatchComponent implements OnInit, OnChanges {
...
+		this.matchService.getFinalFromChampionship(this.match.championship._id).subscribe(
+			(res) => {
+				this.checkFinals(res);
+			},
+			(error : any) => {
+				console.log(error);
+				this.error = error;
+			});
+
 		if (!this.validate()) {
```
Comment (aquintiliano) [inline Frontend/src/app/insert-match/insert-match.component.ts]: you're maybe calling this.validate, which check for final, before the server response, what give you a false positive, use promises to achieve what you want

Comment (cristiadu) [reply Frontend/src/app/insert-match/insert-match.component.ts]: Yeah, I thought about that. I'm also getting some errors from this PR while saving matches. It seems like the id from the current championship takes a while to be set as well, and it's not always rendered.

Comment (cristiadu) [reply Frontend/src/app/insert-match/insert-match.component.ts]: Will take a closer look tomorrow on this. Thank you for the feedback!

**Comment by cristiadu** [comment]
Changed to use promise instead of observable.

**Pull request merged**